packages/firestore/ios/RNFBFirestore/RNFBFirestoreCollectionModule.m

```diff
@@ -216,6 +216,80 @@ - (void)invalidate {
                  }];
 }
 
+RCT_EXPORT_METHOD(aggregateQuery
+                  : (FIRApp *)firebaseApp
+                  : (NSString *)databaseId
+                  : (NSString *)path
+                  : (NSString *)type
+                  : (NSArray *)filters
+                  : (NSArray *)orders
+                  : (NSDictionary *)options
+                  : (NSArray *)aggregateQueries
+                  : (RCTPromiseResolveBlock)resolve
+                  : (RCTPromiseRejectBlock)reject) {
+  FIRFirestore *firestore = [RNFBFirestoreCommon getFirestoreForApp:firebaseApp
+                                                         databaseId:databaseId];
+  FIRQuery *query = [RNFBFirestoreCommon getQueryForFirestore:firestore path:path type:type];
+
+  NSMutableArray<FIRAggregateField *> *aggregateFields =
+      [[NSMutableArray<FIRAggregateField *> alloc] init];
+
+  for (NSDictionary *aggregateQuery in aggregateQueries) {
+    NSString *aggregateType = aggregateQuery[@"aggregateType"];
+    NSString *fieldPath = aggregateQuery[@"field"];
+
+    if ([aggregateType isEqualToString:@"count"]) {
+      [aggregateFields addObject:[FIRAggregateField aggregateFieldForCount]];
+    } else if ([aggregateType isEqualToString:@"sum"]) {
+      [aggregateFields addObject:[FIRAggregateField aggregateFieldForSumOfField:fieldPath]];
+    } else if ([aggregateType isEqualToString:@"average"]) {
+      [aggregateFields addObject:[FIRAggregateField aggregateFieldForAverageOfField:fieldPath]];
+    } else {
+      NSString *reason = [@"Invalid Aggregate Type: " stringByAppendingString:aggregateType];
+      [RNFBFirestoreCommon
+          promiseRejectFirestoreException:reject
+                                    error:[NSException exceptionWithName:
+                                                           @"RNFB Firestore: Invalid Aggregate Type"
+                                                                  reason:reason
+                                                                userInfo:nil]];
+    }
```

Comment:
Another classic - the promise rejection doesn't actually stop native execution, so if you don't return after it, then execution continues, something else blows up and another rejection is attempted (triggering an error where promise was attempted to reject twice....) or somehow it succeeds then the resolve block is also called (another error)

I only know this because of course I have made this mistake before then painfully diagnosed + fixed it post-release some long time ago...


```suggestion
    } else {
      NSString *reason = [@"Invalid Aggregate Type: " stringByAppendingString:aggregateType];
      [RNFBFirestoreCommon
          promiseRejectFirestoreException:reject
                                    error:[NSException exceptionWithName:
                                                           @"RNFB Firestore: Invalid Aggregate Type"
                                                                  reason:reason
                                                                userInfo:nil]];
      return;
    }
```